content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -19,6 +19,79 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
 
 <div class="release-feature-authors">authors: @author</div>
 
+## One-Shot Systems
+
+<div class="release-feature-authors">authors: @alice-i-cecile @pascualex, @Trashtalk217, @Zeenobit</div>
+
+Ordinarily, systems run once per frame, as part of a schedule.
+But what if you don't want to poll for them regularly?
```

Comment:
Hmm yeah, wording isn't clear. I'm trying to hint at there being two different ways to respond to events: push-based and pull-based.

In a push-based idiom, you read the event stream, and then dispatch work based on the events read. This is one-shot systems.

In a pull-based idiom, you have a bunch of listeners that are constantly checking (polling) for changes that they care about. This is ordinary scheduled systems.

Unfortunately that's a terrible explanation for this section of the post.

Reply:
I was originally careful with wording it like 'being push-based' over 'being pull-based'. Since whenever you talk about: I want to run a system based on a rare event, the obvious thing to use is Events. And that would start a whole discussion on whether one-shot systems should be used rather then Events, and I didn't want to get into that.